src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,873 @@
+				for (var i = 0; i < size; i++)
+				{
+					var childNode = value.Children[i];
+					offset++;
+					AddNodeToView(childNode, offset + index);
+					offset = AddNodeDescendantsToView(childNode, index, offset);
+				}
+
+				return offset;
+			}
+
+			return offset;
+		}
+
+		private void RemoveNodeAndDescendantsFromView(TreeViewNode value)
+		{
+			if (value.IsExpanded)
+			{
+				int size = value.Children.Count;
+				for (int i = 0; i < size; i++)
+				{
+					var childNode = (TreeViewNode)value.Children[i];
+					RemoveNodeAndDescendantsFromView(childNode);
+				}
+			}
+
+			bool containsValue = IndexOfNode(value, out var valueIndex);
+			if (containsValue)
+			{
+				RemoveAt(valueIndex);
+			}
+		}
+
+		//void ViewModel.RemoveNodesAndDescendentsWithFlatIndexRange(unsigned int lowIndex, unsigned int highIndex)
+		//{
+		//	MUX_ASSERT(lowIndex <= highIndex);
+
+		//	for (int i = static_cast<int>(highIndex); i >= static_cast<int>(lowIndex); i--)
+		//	{
+		//		RemoveNodeAndDescendantsFromView(GetNodeAt(i));
+		//	}
+		//}
+
+		//int ViewModel.GetNextIndexInFlatTree(const TreeViewNode& node)
+		//{
+		//	unsigned int index = 0;
+		//	bool isNodeInFlatList = IndexOfNode(node, index);
+
+		//	if (isNodeInFlatList)
+		//	{
+		//		index++;
+		//	}
+		//	else
+		//	{
+		//		// node is Root node, so next index in flat tree is 0
+		//		index = 0;
+		//	}
+		//	return index;
+		//}
+
+		//// When ViewModel receives a event, it only includes the sender(parent TreeViewNode) and index.
+		//// We can't use sender[index] directly because it is already updated/removed
+		//// To find the removed TreeViewNode:
+		////   calcuate allOpenedDescendantsCount in sender[0..index-1] first
+		////   then add offset and finally return TreeViewNode by looking up the flat tree.
+		//TreeViewNode ViewModel.GetRemovedChildTreeViewNodeByIndex(TreeViewNode const& node, unsigned int childIndex)
+		//{
+		//	unsigned int allOpenedDescendantsCount = 0;
+		//	for (unsigned int i = 0; i < childIndex; i++)
+		//	{
+		//		TreeViewNode calcNode = node.Children().GetAt(i).as< TreeViewNode > ();
+		//		if (calcNode.IsExpanded())
+		//		{
+		//			allOpenedDescendantsCount += GetExpandedDescendantCount(calcNode);
+		//		}
+		//	}
+
+		//	unsigned int childIndexInFlatTree = GetNextIndexInFlatTree(node) + childIndex + allOpenedDescendantsCount;
+		//	return GetNodeAt(childIndexInFlatTree);
+		//}
+
+		//int ViewModel.CountDescendants(const TreeViewNode& value)
+		//{
+		//	int descendantCount = 0;
+		//	unsigned int size = value.Children().Size();
+		//	for (unsigned int i = 0; i < size; i++)
+		//	{
+		//		var childNode = value.Children().GetAt(i).as< TreeViewNode > ();
+		//		descendantCount++;
+		//		if (childNode.IsExpanded())
+		//		{
+		//			descendantCount = descendantCount + CountDescendants(childNode);
+		//		}
+		//	}
+
+		//	return descendantCount;
+		//}
+
+		private uint IndexOfNextSibling(TreeViewNode childNode)
+			{
+				//	var parentNode = childNode.Parent();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)